mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -1722,25 +1723,22 @@ private class LogisticAggregator(
     var maxMargin = Double.NegativeInfinity
 
     val margins = new Array[Double](numClasses)
+    val featureStdArray = new Array[Double](features.size)
```

Comment:
I suggest change the `featureStdArray` as Aggregator class member, so that avoid each update allocate a new temporary array.